packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -0,0 +1,413 @@
+    DC2_LARGE = "dc2.large",
+    DC2_8XLARGE = "dc2.8xlarge",
+    RA3_16XLARGE = "ra3.16xlarge",
+}
+
+/**
+ * What cluster type to use
+ */
+export enum ClusterType {
+    SINGLE_NODE = "single-node",
+    MULTI_NODE = "multi-node"
+}
+
+/**
+ * Properties for a new database cluster
+ */
+export interface ClusterProps {
+
+    /**
+     * An optional identifier for the cluster
+     *
+     * @default - A name is automatically generated.
+     */
+    readonly clusterIdentifier?: string;
+
+    /**
+     * Additional parameters to pass to the database engine
+     * https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html
+     *
+     * @default - No parameter group.
+     */
+    readonly parameterGroup?: IParameterGroup;
+
+    /**
+     * Number of compute nodes in the cluster
+     *
+     * Value must be at least 1 and no more than 100.
+     *
+     * @default 1
+     */
+    readonly numberOfNodes?: number;
+
+    /**
+     * The node type to be provisioned for the cluster.
+     *
+     * @default NodeType.DC2_LARGE
+     */
+    readonly nodeType?: NodeType;
+
+    /**
+     * Settings for the individual instances that are launched
+     *
+     * @default ClusterType.MULTI_NODE
+     */
+    readonly clusterType?: ClusterType;
+
+    /**
+     * What port to listen on
+     *
+     * @default - The default for the engine is used.
+     */
+    readonly port?: number;
+
+    /**
+     * Whether to enable encryption of data in the cluster
+     *
+     * @default false
+     */
+    readonly encrypted?: boolean
+
+    /**
+     * The KMS key for storage encryption.
+     * will be set to `true`.
+     *
+     * @default - default master key.
+     */
+    readonly kmsKey?: kms.IKey;
+
+    /**
+     * A preferred maintenance window day/time range. Should be specified as a range ddd:hh24:mi-ddd:hh24:mi (24H Clock UTC).
+     *
+     * Example: 'Sun:23:45-Mon:00:15'
+     *
+     * @default - 30-minute window selected at random from an 8-hour block of time for
+     * each AWS Region, occurring on a random day of the week.
+     * @see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_UpgradeDBInstance.Maintenance.html#Concepts.DBMaintenance
+     */
+    readonly preferredMaintenanceWindow?: string;
+
+    /**
+     * The VPC to place the cluster in.
+     *
+     * @default a new Vpc will be created
+     */
+    readonly vpc?: ec2.IVpc;
+
+    /**
+     * Security group.
+     *
+     * @default a new security group is created.
+     */
+    readonly securityGroups?: ec2.ISecurityGroup[];
+
+    /**
+     * Username and password for the administrative user
+     */
+    readonly masterUser: Login;
+
+    /**
+     * A list of AWS Identity and Access Management (IAM) roles that can be used by the cluster to access other AWS services.
+     * Specify a maximum of 10 roles.
+     *
+     * @default - A role is automatically created for you
+     */
+    readonly iamRoles?: IRole[];
```

Comment:
1. What's the meaning of passing multiple roles here? I'm genuinely asking (I've only ever seen a single role passed to resources that need it).
2. We usually skip the service name prefix. So this should just be `roles`.

Reply:
> 1. What's the meaning of passing multiple roles here? I'm genuinely asking (I've only ever seen a single role passed to resources that need it).

I thought it would model the way how Redshift does roles (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-redshift-cluster.html#cfn-redshift-cluster-iamroles). You can add up to 10 roles and use them when doing queries.

The alternative would be some kind of method that allows you to attach additional roles.
What is your take on that?

Thinking about your question it could also be reasonable to just `@default` to no role at all. However I think it be a nicer dev experience if one was provided from the start. :thinking:

Reply:
Maybe I don't understand how Redshift works. Can you explain:

1. What does it mean to have more than one role associated with a cluster?
2. What doe it mean to have 0 roles associated with the cluster?

Reply:
Roles are a means for Redshift to access other AWS services (like S3, Glue or DynamoDB) when using the `UNLOAD` or `COPY` commands (https://docs.aws.amazon.com/redshift/latest/mgmt/authorizing-redshift-service.html). During `UNLOAD` or `COPY` you can pass an IAM role associated with the cluster for authorization like this:

```sql
copy catdemo
from 's3://awssampledbuswest2/tickit/category_pipe.txt'
iam_role 'arn:aws:iam::<aws-account-id>:role/<role-name>'
region 'us-west-2';
```

_from https://docs.aws.amazon.com/redshift/latest/dg/r_COPY.html_

1. You could have two roles attached to the cluster. One allowing access to S3, one allowing access to DynamoDB for example.
2. Redshift cannot access other AWS services